lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -800,19 +800,24 @@ class ClassMetadataInfo implements ClassMetadata
     /** @var InstantiatorInterface|null */
     private $instantiator;
 
+    /** @var array<class-string|string,class-string<Type>|string> $typedFieldMappings */
+    private $typedFieldMappings;
+
     /**
      * Initializes a new ClassMetadata instance that will hold the object-relational mapping
      * metadata of the class with the given name.
      *
-     * @param string $entityName The name of the entity class the new instance is used for.
+     * @param string                                                $entityName         The name of the entity class the new instance is used for.
+     * @param array<class-string|string, class-string<Type>|string> $typedFieldMappings
```

Comment:
is there a way to make some custom psalm-type which would only allow subset of strings? What Id like is to make e.g. `scalar-name` which would only allow 'int','string','float','bool','array'. This way we could replace with more precise
```php
@param array<class-string|scalar-name, class-string<Type>|string> $typedFieldMappings
```

Reply:
Yes I think you can use `'int'|'string'|'float'` etc. in your type

Reply:
We cannot restrict the type names there. Custom types are registered with a name too so the list of available strings is infinite.

Reply:
@stof this is PHP types, not DBAL types